sdk/storage/azure-storage-file-share/tests/test_share.py

```diff
@@ -184,6 +184,232 @@ def test_undelete_share(self, resource_group, location, storage_account, storage
+        share_client = self._create_share('test')
+
+        # Act
+        proposed_lease_id = '55e97f64-73e8-4390-838d-d9e84a374321'
+        lease = share_client.acquire_lease(lease_id=proposed_lease_id)
+
+        # Assert
+        self.assertEqual(proposed_lease_id, lease.id)
+
+    @GlobalStorageAccountPreparer()
+    def test_lease_share_change_lease_id(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        share_client = self._create_share('test')
+
+        # Act
+        lease_id = '29e0b239-ecda-4f69-bfa3-95f6af91464c'
+        lease = share_client.acquire_lease()
+        lease_id1 = lease.id
+        lease.change(proposed_lease_id=lease_id)
+        lease.renew()
+        lease_id2 = lease.id
+
+        # Assert
+        self.assertIsNotNone(lease_id1)
+        self.assertIsNotNone(lease_id2)
+        self.assertNotEqual(lease_id1, lease_id)
+        self.assertEqual(lease_id2, lease_id)
+
+    @GlobalStorageAccountPreparer()
+    def test_set_share_metadata_with_lease_id(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        share_client = self._create_share('test')
+        metadata = {'hello': 'world', 'number': '43'}
+        lease_id = share_client.acquire_lease()
+
+        # Act
+        share_client.set_share_metadata(metadata, lease=lease_id)
+
+        # Assert
+        md = share_client.get_share_properties().metadata
+        self.assertDictEqual(md, metadata)
+
+    @GlobalStorageAccountPreparer()
+    def test_get_share_metadata_with_lease_id(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        share_client = self._create_share('test')
+        metadata = {'hello': 'world', 'number': '43'}
+        share_client.set_share_metadata(metadata)
+        lease_id = share_client.acquire_lease()
+
+        # Act
+        md = share_client.get_share_properties(lease=lease_id).metadata
+
+        # Assert
+        self.assertDictEqual(md, metadata)
+
+    @GlobalStorageAccountPreparer()
+    def test_get_share_properties_with_lease_id(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        share_client = self._create_share('test')
+        metadata = {'hello': 'world', 'number': '43'}
+        share_client.set_share_metadata(metadata)
+        lease_id = share_client.acquire_lease()
+
+        # Act
+        props = share_client.get_share_properties(lease=lease_id)
+        lease_id.break_lease()
+
+        # Assert
+        self.assertIsNotNone(props)
+        self.assertDictEqual(props.metadata, metadata)
+        self.assertEqual(props.lease.duration, 'infinite')
+        self.assertEqual(props.lease.state, 'leased')
+        self.assertEqual(props.lease.status, 'locked')
```

Comment:
Can we add a test for list blob like this to make sure the lease is parsed properly when calling from_generated()

Reply:
I'm not sure I follow, this is a test for share leases not blobs?

Reply:
I have added two more tests for listing shares and ensuring that lease is being parsed as a share property